src/turing_inference.jl

```diff
@@ -39,39 +39,32 @@ function turing_inference(prob::DiffEqBase.DEProblem,alg,t,data,priors;
     _saveat = t === nothing ? Float64[] : t
     sol_tmp = solve(p_tmp, alg; saveat = _saveat, kwargs...)
 
-    if sol_tmp isa DiffEqBase.AbstractNoTimeSolution
-      res = sol_tmp.u
-      __lp = Turing.observe(
-        sampler,
-        likelihood(res,theta,Inf,σ),   # Distribution
-        x,    # Data point
-        vi
-      )
-      vi.logp += __lp
+    if any((s.retcode != :Success for s in sol_tmp)) && any((s.retcode != :Terminated for s in sol_tmp))
+      vi.logp += Inf
```

Comment:
Should this not be `-Inf`? But I might easily misunderstand what you are trying to do.